packages/fxa-auth-server/scripts/delete-unverified-accounts/delete-unverified-accounts.ts

```diff
@@ -0,0 +1,308 @@
+        continue;
+      }
+      const result = await accountTasks.deleteAccount({
+        uid: acct.uid,
+        customerId: (await getAccountCustomerByUid(acct.uid))?.stripeCustomerId,
+        reason,
+      });
+      console.log(`Created cloud task ${result} for uid ${x}`);
+    }
+  }
+
+  if (useDateRange) {
+    if (program.tableScan !== 'true') {
+      console.log('Please call with --table-scan if you are sure.');
+      return 0;
+    }
+
+    await deleteAccountInDateRange(program.startDate, program.endDate, {
+      taskLimit,
+      reason,
+      accountTasks,
+      auth: config.auth,
+    });
+  }
+
+  return 0;
+};
+
+/**
+ * Enqueue account deletes to Cloud Task with a date range.
+ *
+ * @param options
+ */
+async function deleteAccountInDateRange(startDate, endDate, options) {
+  const taskLimit = options.taskLimit;
+  const reason = options.reason;
+  const accountTasks = options.accountTasks;
+  const authDbConfig = options.auth;
+  const tableScan = options.tableScan;
+
+  if (tableScan !== 'true') {
+    console.log('Please call with --table-scan if you are sure.');
+    return 0;
+  }
+
+  const kyselyDb = await setupAccountDatabase(authDbConfig);
+  const accounts = await kyselyDb
+    .selectFrom('accounts')
+    .where('accounts.emailVerified', '=', 0)
+    .where('accounts.createdAt', '>=', startDate)
+    .where('accounts.createdAt', '<=', endDate)
+    .leftJoin('accountCustomers', 'accounts.uid', 'accountCustomers.uid')
+    .select(['accounts.uid', 'accountCustomers.stripeCustomerId'])
+    .execute();
+
+  // Scaling suggestion is 500/5/50 rule, may start at 500/sec, and increase every 5 minutes by 50%.
+  // They also note increased latency may occur past 1000/sec, so we stop increasing as we approach that.
+  const scaleUpIntervalMins = 5;
+  let lastScaleUp = Date.now();
+  let rateLimit = taskLimit;
+  const queue = new PQueue({
+    interval: 1000,
+    intervalCap: rateLimit,
+    concurrency: rateLimit * 2,
+  });
+
+  for (const row of accounts) {
+    if (
+      rateLimit < 950 &&
+      Date.now() - lastScaleUp > scaleUpIntervalMins * 60 * 1000
+    ) {
+      rateLimit = Math.floor(rateLimit * 1.5);
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      queue['#intervalCap'] = rateLimit; // This is private, but we need to update it
+      queue.concurrency = rateLimit * 2;
+      lastScaleUp = Date.now();
+    }
+
+    await queue.onSizeLessThan(rateLimit * 4); // Back-pressure
+
+    queue.add(async () => {
+      try {
+        const result = await accountTasks.deleteAccount({
+          uid: row.uid.toString('hex'),
+          customerId: row.stripeCustomerId || undefined,
+          reason,
+        });
+        console.log(
+          `Created cloud task ${result} for uid ${row.uid.toString('hex')}`
+        );
+      } catch (err) {
+        console.error('Errored creating task', err);
```

Comment:
If something gets `catch`ed here, does it still set the `exitCode` to 1+ so the script fails? Or is a `console.error()` good enough here?